pkg/actors/internal_actor.go

```diff
@@ -21,145 +21,57 @@ import (
 	"encoding/json"
 	"fmt"
 	"io"
-	"strings"
 
-	"github.com/dapr/dapr/pkg/apphealth"
-	"github.com/dapr/dapr/pkg/config"
-	invokev1 "github.com/dapr/dapr/pkg/messaging/v1"
-	commonv1pb "github.com/dapr/dapr/pkg/proto/common/v1"
+	"github.com/dapr/dapr/pkg/actors/internal"
 )
 
 const InternalActorTypePrefix = "dapr.internal."
 
 // InternalActor represents the interface for invoking an "internal" actor (one which is built into daprd directly).
 type InternalActor interface {
 	SetActorRuntime(actorsRuntime Actors)
-	InvokeMethod(ctx context.Context, actorID string, methodName string, data []byte) (any, error)
+	InvokeMethod(ctx context.Context, actorID string, methodName string, data []byte, metadata map[string][]string) ([]byte, error)
```

Comment:
Why do we need extra `metadata` here ?

Reply:
Good question, let me see if I can get away with removing it

Reply:
The metadata includes things like headers and re-entrancy IDs. I think it may be safe to leave it there in case an internal actor needs them. I don't believe it's the case today, but probably not a bad practice